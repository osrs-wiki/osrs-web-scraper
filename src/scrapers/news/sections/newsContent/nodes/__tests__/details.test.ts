import { MediaWikiBuilder } from "@osrs-wiki/mediawiki-builder";
import parse from "node-html-parser";

import detailsParser from "../details";

describe("summary node", () => {
  test("summary elements should parse and render", () => {
    const root = parse("<details><summary>test</summary><b>test</b></details>");
    const builder = new MediaWikiBuilder();
    builder.addContents([detailsParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });
});
