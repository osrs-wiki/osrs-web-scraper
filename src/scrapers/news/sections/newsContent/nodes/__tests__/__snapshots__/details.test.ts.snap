// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`summary node summary elements should parse and render 1`] = `
"{{Collapsed section|18|test}}
'''test'''{{Collapsed section end}}
"
`;
